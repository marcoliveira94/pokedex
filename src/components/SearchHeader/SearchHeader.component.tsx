import React from 'react';
import styled from 'styled-components';
import {Animated, TextInput} from 'react-native';

interface SearchHeaderProps {
  onSearch: (text: string) => void;
  value?: string;
  color: string;
}

const SearchHeader = ({onSearch, value, color}: SearchHeaderProps) => {
  return (
    <SearchWrapper color={color}>
      <SearchInput
        testID={'search-header-input'}
        onChangeText={onSearch}
        value={value}
      />
    </SearchWrapper>
  );
};

export default SearchHeader;

const SearchWrapper = styled(Animated.View)<{color: string}>`
  align-self: stretch;
  text-align: center;
  background-color: ${(props) => props.color};
  align-items: center;
  justify-content: center;
  border-bottom-left-radius: 20px;
  border-bottom-right-radius: 20px;
  height: 200px;
`;

const SearchInput = styled(TextInput)`
  width: 80%;
  height: 40px;
  margin: 20px;
  background-color: white;
  border-radius: 20px;
  padding: 10px;
`;
