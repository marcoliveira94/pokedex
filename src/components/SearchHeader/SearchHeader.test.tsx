import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import SearchHeader from './SearchHeader.component';

jest.mock('../../shared/hooks/useDominantColor.hook', () => ({
  useDominantColor: () => ({
    color: {
      background: 'black',
      detail: 'black',
      platform: undefined,
      primary: 'black',
      secondary: 'black',
    },
    error: null,
  }),
}));

describe('SearchHeader', () => {
  const onSearch = jest.fn();

  it('renders correctly', () => {
    const tree = renderer
      .create(<SearchHeader onSearch={onSearch} value={'test'} color="red" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('onSearch', () => {
    const {getByTestId} = render(
      <SearchHeader onSearch={onSearch} value={'test'} color="red" />,
    );
    const input = getByTestId('search-header-input');
    fireEvent(input, 'onChangeText', 'ab');
    expect(onSearch).toHaveBeenCalledWith('ab');
  });
});
