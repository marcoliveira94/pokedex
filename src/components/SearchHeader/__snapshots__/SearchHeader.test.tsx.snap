// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchHeader renders correctly 1`] = `
<View
  className="sc-AxjAm bUYfNO"
  color="red"
>
  <TextInput
    allowFontScaling={true}
    className="sc-AxirZ ksCyHq"
    onChangeText={[MockFunction]}
    rejectResponderTermination={true}
    testID="search-header-input"
    underlineColorAndroid="transparent"
    value="test"
  />
</View>
`;
